tilecloud_chain/__init__.py

```diff
@@ -1336,10 +1337,13 @@ def __call__(self, tile):
                     "z": tile.tilecoord.z,
                 }
                 logger.debug("[%s] process: %s", tile.tilecoord, command)
-                code = subprocess.call(command, shell=True)
-                if code != 0:  # pragma: no cover
-                    tile.error = "Command '{}' on tile {} return error code {}".format(
-                        command, tile.tilecoord, code
+                process = subprocess.Popen(
+                    command, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE
+                )
+                output, err = process.communicate()
```

Comment:
Seems it could use `subprocess.run`:
```
result = subprocess.run(command, shell=True, capture_output=True)
if result.returncode !=0:
    tile.error = "Command '{}' on tile {} return error code {}:\n{}\n{}".format(
        command, tile.tilecoord, result.returncode, result.stderr, result.stdout
    )
```
or
```
try:
  subprocess.run(command, shell=True, check=True, capture_output=True)
except CalledProcessError as e:
    tile.error = "Command '{}' on tile {} return error code {}:\n{}\n{}".format(
        command, tile.tilecoord, e.returncode, e.stderr, e.stdout
    )
```

Reply:
Thanks, updated with solution 1 :-)